src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsalertsapi/service/AlertCodeService.kt

```diff
@@ -55,6 +55,20 @@ class AlertCodeService(
       }
     }
 
+  @Transactional
+  fun reactivateAlertCode(alertCode: String, alertRequestContext: AlertRequestContext): AlertCode =
+    alertCode.let {
+      it.checkAlertCodeExists()
+      with(alertCodeRepository.findByCode(it)!!) {
+        if (deactivatedAt != null) {
+          deactivatedAt = null
+          deactivatedBy = null
+          reactivate(alertRequestContext.requestAt)
+        }
+        alertCodeRepository.saveAndFlush(this).toAlertCodeModel()
```

Comment:
You could move this into the if statement so that it doesn't run save and flush unnecessarily